plans/action/install.pp

```diff
@@ -306,55 +312,52 @@
+    $common_install_flags = [
       '--puppet-service-ensure', 'stopped',
       "main:dns_alt_names=${dns_alt_names_csv}",
-
-      # Params that vary depending on what kind of server this is.
-      # The undef values will be compacted out
       "main:certname=${target.peadm::target_name()}",
-      ($target in $compiler_a_targets) ? {
-        false => undef,
-        true  => [
+    ]
+
+    if ($target in $compiler_a_targets) {
+      run_task('peadm::agent_install', $target,
+        server        => $master_target.peadm::target_name(),
+        install_flags => $common_install_flags + [
           "extension_requests:${peadm::oid('pp_auth_role')}=pe_compiler",
           "extension_requests:${peadm::oid('peadm_availability_group')}=A",
         ],
-      },
-      ($target in $compiler_b_targets) ? {
-        false => undef,
-        true  => [
+      )
+    }
+    elsif ($target in $compiler_b_targets) {
+      run_task('peadm::agent_install', $target,
+        server        => $master_target.peadm::target_name(),
+        install_flags => $common_install_flags + [
           "extension_requests:${peadm::oid('pp_auth_role')}=pe_compiler",
           "extension_requests:${peadm::oid('peadm_availability_group')}=B",
         ],
-      },
-      ($target in $master_replica_target) ? {
-        false => undef,
-        true => [
-          "extension_requests:${peadm::oid('peadm_availability_group')}=B",
+      )
+    }
+    elsif ($target in $master_replica_target) {
+      run_task('peadm::agent_install', $target,
+        server        => $master_target.peadm::target_name(),
+        install_flags => $common_install_flags + [
           "extension_requests:${peadm::oid('peadm_role')}=puppet/master",
+          "extension_requests:${peadm::oid('peadm_availability_group')}=B",
         ],
-      },
-
-    ])
-  }}
-
-  # Ensure certificate requests have been submitted
-  run_task('peadm::submit_csr', $agent_installer_targets)
-
-  # TODO: come up with an intelligent way to validate that the expected CSRs
-  # have been submitted and are available for signing, prior to signing them.
-  # For now, waiting a short period of time is necessary to avoid a small race.
-  ctrl::sleep(15)
+      )
+    }
 
-  run_task('peadm::sign_csr', $master_target,
-    certnames => $agent_installer_targets.map |$target| { $target.name },
-  )
+    # Ensure certificate requests have been submitted
+    run_task('peadm::submit_csr', $target)
+    # TODO: come up with an intelligent way to validate that the expected CSRs
+    # have been submitted and are available for signing, prior to signing them.
+    # For now, waiting a short period of time is necessary to avoid a small race.
+    ctrl::sleep(5)
+    run_task('peadm::sign_csr', $master_target, { 'certnames' => [$target.name] } )
+    run_task('peadm::puppet_runonce', $target)
+  }
 
-  run_task('peadm::puppet_runonce', $all_targets - $master_target)
+  run_task('peadm::puppet_runonce', $database_targets )
```

Comment:
Future reference note: we should check and see if this Puppet run actually makes any changes. If it doesn't, we can remove it. I think it might be redundant.